cmd/stack_lint.go

```diff
@@ -0,0 +1,126 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var lintCmd = &cobra.Command{
+	Use:   "lint",
+	Short: "Lint your stack to verify that it conforms to the standard of an Appsody stack",
+	Long: `This command will validate that your stack has the structure of an Appsody stack. It will inform you of files/directories
+missing and warn you if your stack could be enhanced.
+
+This command can be run from the base directory of your stack or you can supply a path to the stack as an argument.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		stackPath, _ := os.Getwd()
+		errorCount := 0
+		warningCount := 0
+
+		if len(args) > 0 {
+			stackPath = args[0]
+		}
+
+		imagePath := filepath.Join(stackPath, "image")
+		templatePath := filepath.Join(stackPath, "/templates")
+		configPath := filepath.Join(imagePath, "/config")
+		projectPath := filepath.Join(imagePath, "/project")
+
+		Info.log("LINTING ", path.Base(stackPath))
+
+		if fileDoesNotExist(filepath.Join(stackPath, "/README.md")) {
+			Error.log("Missing README.md in: ", stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(filepath.Join(stackPath, "/stack.yaml")) {
+			Error.log("Missing stack.yaml in: ", stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(imagePath) {
+			Error.log("Missing image directory in ", stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(filepath.Join(imagePath, "/Dockerfile-stack")) {
+			Error.log("Missing Dockerfile-stack in ", imagePath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(filepath.Join(imagePath, "/LICENSE")) {
+			Error.log("Missing LICENSE in ", imagePath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(configPath) {
+			Warning.log("Missing config directory in ", imagePath, " (Knative deployment will be used over Kubernetes)")
+			warningCount++
+
+		}
+
+		if fileDoesNotExist(filepath.Join(configPath, "/app-deploy.yaml")) {
+			Warning.log("Missing app-deploy.yaml in ", configPath, " (Knative deployment will be used over Kubernetes)")
+			warningCount++
+		}
+
+		if fileDoesNotExist(filepath.Join(projectPath, "/Dockerfile")) {
+			Warning.log("Missing Dockerfile in ", projectPath)
+			warningCount++
+		}
+
+		if fileDoesNotExist(templatePath) {
+			Error.log("Missing template directory in: ", stackPath)
+			errorCount++
+		}
+
+		if IsEmptyDir(templatePath) {
+			Error.log("No templates found in: ", templatePath)
+			errorCount++
+		}
+
+		templates, _ := ioutil.ReadDir(templatePath)
+		for _, f := range templates {
+			if !fileDoesNotExist(filepath.Join(templatePath, f.Name(), ".appsody-config.yaml")) {
+				Error.log("Unexpected .appsody-config.yaml in ", filepath.Join(templatePath, f.Name()))
+				errorCount++
+			}
+		}
+
+		if errorCount > 0 {
```

Comment:
I think you could print the total errors even if it is 0. Here's what I'm thinking to finish the linter:
```
Info.log("TOTAL ERRORS: ", errorCount)
Info.log("TOTAL WARNINGS: ", warningCount)
if errorCount > 0 {
	return errors.Errorf("LINT TEST FAILED")
} else {
	Info.log("LINT TEST PASSED")
	return nil
}
```